src/MonitoringFunctions/IDataService.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace MonitoringFunctions
+{
+    internal interface IDataService : IDisposable
```

Comment:
Why is IDisposable used here? It is for possible future implementations?

Reply:
Yes. I find it likely that a data service we will implement in the future will have some long living object that needs to be disposed. If I don't include this in the interface today, it will be harder to update the code to call `Dispose` method at correct places. I may be wrong and this may never be used. I just chose the option that looks safer.